assets/js/components/StatesBarChart.vue

```diff
@@ -365,17 +365,52 @@ export default Chart.extend({
         }, {})
     },
 
+    /**
+     * returns projects/programmes for donors/beneficiaries
+     * if a donor or beneficiary is selected (filter is applied)
+     * it will show the projects/programmes for that state
+     */
     legend_items() {
       // just the totals merged into the types
       const out = {}
+
       for (const t in this.types) {
-        out[t] = Object.assign({
-          value: this.totals[t],
-        }, this.types[t])
+        if(this.filters[this.state_type]) {
+          out[t] = Object.assign({
+            value: this.filter_values[t],
+          }, this.types[t])
+        } else {
+          out[t] = Object.assign({
+            value: this.totals[t],
+          }, this.types[t])
+        }
       }
+
       // return an array though
       return d3.values(out)
     },
+
+    /**
+     * finds the selected state (beneficiary or donor) returns its programmes/projects
+     * @return {Object} projects
+     * @return {number} projects.programmes
+     * @return {number} projects.projects
+     */
+    filter_values(){
```

Comment:
this function is a fancy way to access `this.aggregated[state_id]`, pre-passed through `valuefunc`.
(note: if you're gonna access this.aggregated from something that's used in a template you need to test for `this.hasData`, or things will explode)

also, there's no point in it being computed, as it isn't passed around as a component property, it's not used in a template, nor does it need to be cached. a method would make more sense, but it's so trivial that inlining it under `legend_items` would just be easier to understand.